pkg/server/server_test.go

```diff
@@ -37,3 +41,44 @@ func TestServerHealthz(t *testing.T) {
 		t.Errorf("expected status code %d, got %d", http.StatusOK, w.Code)
 	}
 }
+
+func TestServerAccessLog(t *testing.T) {
+	var buff bytes.Buffer
+	config := zap.NewProductionConfig()
+	// construct core to make it use the buffer
+	core := zapcore.NewCore(
+		zapcore.NewJSONEncoder(config.EncoderConfig),
+		zapcore.AddSync(&buff),
+		config.Level,
+	)
+	logger := zap.New(core)
+	zapLogger = logger
+	server, err := NewServer("localhost", 8080, false)
```

Comment:
Rather than having a global `zapLogger`, can we create an internal "constructor" that includes the logger as an argument?

Ex:

```go
func NewServer(...) (*SprayProxyServer, error) {
    return newServerWithLogger(logger.Get(), ...)
}

Reply:
That's neat and it allows me to override the logger in test. The second reason I used the global though is that the logger should be flushed on exit (e.g `logger, _ := zap.NewProduction(); defer logger.Sync() // flushes buffer, if any`) and for that I need a long running function like Run() and not a short running one as a constructor. With the global I'm able to do https://github.com/enarha/sprayproxy/blob/add-access-log/pkg/server/server.go#L65 which is ugly, but I couldn't think of something better.

Reply:
Makes sense. I've added this to the scope of https://issues.redhat.com/browse/PLNSRVCE-1081 (graceful termination). I think the refactors needed to make graceful termination feasible will help in this regard.